disnake/i18n.py

```diff
@@ -0,0 +1,232 @@
+def is_valid_locale(locale: str) -> bool:
+    """
+    Returns ``True`` if the locale is valid for use with the API, ``False`` otherwise.
+
+    .. versionadded:: 2.5
+    """
+    return locale in Locale._enum_value_map_  # type: ignore
+
+
+class LocalizationValue:
+    """
+    Internal container type for (pending) localization data
+
+    .. versionadded:: 2.5
+    """
+
+    __slots__ = ("_key", "_data")
+
+    def __init__(self, localizations: Optional[Localizations]):
+        self._key: Optional[str]
+        self._data: Optional[Dict[str, str]]
+
+        if localizations is None:
+            # no localization
+            self._key = None
+            self._data = None
+        elif isinstance(localizations, str):
+            # got localization key
+            self._key = localizations
+            self._data = MISSING  # not localized yet
+        elif isinstance(localizations, dict):
+            # got localization data
+            self._key = None
+            self._data = {str(k): v for k, v in localizations.items()}
+        else:
+            raise TypeError(f"Invalid localizations type: {type(localizations).__name__}")
+
+    def _link(self, store: LocalizationStore) -> None:
+        """Loads localizations from the specified store if this object has a key"""
+        if self._key is not None:
+            self._data = store.get(self._key)
+
+    def to_dict(self) -> Optional[Dict[str, str]]:
+        """Returns a dict with a locale -> localization mapping"""
+        if self._data is MISSING:
+            warnings.warn(
+                "value was never localized, this is likely a library bug",
+                LocalizationWarning,
+                stacklevel=2,
+            )
+            return None
+        return self._data
+
+    def __eq__(self, other) -> bool:
+        d1 = self.to_dict()
+        d2 = other.to_dict()
+        # consider values equal if they're both falsy, or actually equal
+        # (it doesn't matter if localizations are `None` or `{}`)
+        return (not d1 and not d2) or d1 == d2
+
+
+class LocalizationStore:
+    """
+    Manages a key-value mapping of localizations
+
+    .. versionadded:: 2.5
+
+    Attributes
+    ------------
+    strict: :class:`bool`
+        Specifies whether :meth:`.get` raises an exception if localizations for a provided key couldn't be found.
+    """
+
+    def __init__(self, *, strict: bool):
+        self.strict = strict
+
+        self._loc: DefaultDict[str, Dict[str, str]] = defaultdict(dict)
+        self._paths: Set[Path] = set()
+
+    def get(self, key: str) -> Optional[Dict[str, str]]:
+        """
+        Returns localizations for the specified key.
+
+        Parameters
+        ----------
+        key: :class:`str`
+            The lookup key.
+
+        Raises
+        ------
+        RuntimeError
+            No localizations for the provided key were found.
+            Raised only if :attr:`strict` is enabled, warns and returns ``None`` otherwise.
```

Comment:
why warn?

Reply:
good point, fixed in 17208e168c1fae5e7be5c08af84679f7d2020a41